compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp

```diff
@@ -1054,15 +1060,47 @@ namespace cppmicroservices
                                 futuresCV.notify_one();
                             }
                         });
+                    if (checkExecuted)
+                    {
+                        bool expected = false;
+                        bool desired = true;
+                        // if asyncStarted is non null AND *asyncStarted==true
+                        if (asyncStarted && !std::atomic_compare_exchange_strong(&(*asyncStarted), &expected, desired))
+                        {
+                            // this is blocking and it has started
+                            return;
+                        }
+                        // else it is non-blocking or it has not started and we now own it
+                    }
                     func();
                 });
-
             std::shared_future<void> fut = task.get_future().share();
+
+            auto taskPtr = std::make_shared<ActualTask>(std::move(task));
+            // this task goes to async queue, we wan't to check if it is already executed
+            PostTask post_task(
+                [taskPtr]() mutable
+                {
+                    try
+                    {
+                        (*taskPtr)(true);
+                    }
+                    catch (...)
+                    {
+                        /*
+                         * task was already executed, by waiting thread, calling it
+                         * again will throw. This is expected, we can just catch and continue
+                         */
+                    }
+                    // get future to propogate real exceptions if any
+                    taskPtr->get_future().share().get();
```

Comment:
huh... not sure why this isn't failing... I added this line as a result of a test where the throw wasn't propagating, and I am guessing the test is now passing becuase this get_future is throwing instead of the correct error... But yes, I'll fix this.